### PR TITLE
render docs for devices.qubit functions

### PR DESCRIPTION
**Pull request by timmysilv**
I am open to notable variation in the docs here, just lmk! I mostly just wanted to get this PR going.

**Context:**
The `devices.qubit` functions are very useful, and people should be able to see them! They're still a little unstable, so I'm inclined to leave those warnings in the docs for now, but they should be visible.

**Description of the Change:**
Add the `devices.qubit` qubit-simulation functions at the bottom of the `qml.devices` sub-page in the "Internal" section (check it out [here](https://xanaduai-pennylane--4555.com.readthedocs.build/en/4555/code/qml_devices.html#module-pennylane.devices.qubit)!)

**Benefits:**
Nice pretty docs to show users all the cool features and helper functions we have

**Possible Drawbacks:**
It's still kinda unstable while we ramp up DQ2, but it should be ready by the time we release 0.33

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -26,6 +26,10 @@
 
 <h3>Documentation 📝</h3>
 
+* Add functions for qubit-simulation to the `qml.devices` sub-page of the "Internal" section.
+  Note that these functions are unstable while device upgrades are underway.
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:30]: This note might be unnecessary by the time 0.33 rolls around.

**Review by mudit2812 — APPROVED** [review]

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (albi3ro) [reply doc/releases/changelog-dev.md:30]: We probably could start to lock them down. They are handy enough that we should think about it, especially for `apply_operation`.

File: pennylane/devices/qubit/simulate.py [hunk 2]
```diff
@@ -116,18 +116,18 @@ def simulate(circuit: qml.tape.QuantumScript, rng=None, debugger=None) -> Result
...
-        debugger (._Debugger): The debugger to use
+        debugger (_Debugger): The debugger to use
 
     Returns:
         tuple(TensorLike): The results of the simulation
 
     Note that this function can return measurements for non-commuting observables simultaneously.
 
-    It does currently not support sampling or observables without diagonalizing gates.
+    It currently does not support observables without diagonalizing gates.
```
Comment (albi3ro) [inline pennylane/devices/qubit/simulate.py:130]:
```suggestion
```

**Review by albi3ro — COMMENTED** [review]
Could we add `adjoint_jacobian`, `adjoint_jvp`, and `adjoint_vjp` to the table of contents as well?

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (timmysilv) [reply doc/releases/changelog-dev.md:30]: I agree that it probably won't be necessary by then. If we stabilize them, I'll remove this line.

**Pull request merged**